outbreak_distribution.R

```diff
@@ -0,0 +1,246 @@
+
+intervals <- lapply(outbreak_list, cut, breaks = breaks)
+prop <- lapply(intervals, function(interval) table(interval) / sum(table(interval)))
+outbreak_size_list <- lapply(prop, as.data.frame)
+for (i in seq_len(nrow(scenarios))) {
+  outbreak_size_list[[i]]$R <- scenarios[i, "R"]
+  outbreak_size_list[[i]]$k <- scenarios[i, "k"]
+  outbreak_size_list[[i]]$offspring_dist <- scenarios[i, "offspring_dist"]
+  outbreak_size_list[[i]]$statistic <- scenarios[i, "statistic"]
+}
+outbreak_size <- do.call(rbind, outbreak_size_list)
+head(outbreak_size)
+
+# Plot outbreak size distribution (Negative binomial) ---------------------
+
+ggplot2::ggplot(data = outbreak_size) +
+  ggplot2::geom_col(
+    mapping = ggplot2::aes(x = as.factor(R), y = Freq, fill = interval)
+  ) +
+  ggplot2::scale_x_discrete(name = "Reproduction number (R)") +
+  ggplot2::scale_y_continuous(name = "Proportion of outbreaks") +
+  ggplot2::scale_fill_brewer(
+    name = "Outbreak size", 
+    palette = "Spectral"
+  ) + 
+  ggplot2::facet_wrap(
+    facets = c("k"), 
+    labeller = ggplot2::label_both
+  ) +
+  ggplot2::theme_bw()
+
+
+# Simulating outbreak sizes for past outbreaks ----------------------------
+
+offspring_dists <- epiparameter::epiparameter_db(
+  epi_name = "offspring distribution"
+)
+
+# Check all empirical distributions are the same --------------------------
+
+vapply(offspring_dists, family, FUN.VALUE = character(1))
+
+# Simulate empirical outbreak size distributions --------------------------
+
+outbreak_list <- vector(mode = "list", length = length(offspring_dists))
+for (i in seq_along(offspring_dists)) {
+  offspring_dist_fun <- match.fun(paste0("r", family(offspring_dists[[i]])))
+  outbreak_list[[i]] <- epichains::simulate_chain_stats(
+    n_chains = n_chains, 
+    statistic = "size", 
+    offspring_dist = offspring_dist_fun,
+    mu = get_parameters(offspring_dists[[i]])[["mean"]],
+    size = get_parameters(offspring_dists[[i]])[["dispersion"]],
+    stat_threshold = breaks[length(breaks) - 1] + 1
+  )
+}
+
+# Group outbreak sizes ----------------------------------------------------
+
+# paste suffix as some diseases have multiple offspring distributions
+diseases <- make.unique(
+  vapply(offspring_dists, `[[`, FUN.VALUE = character(1), "disease")
+)
+
+intervals <- lapply(outbreak_list, cut, breaks = breaks)
+prop <- lapply(intervals, function(interval) table(interval) / sum(table(interval)))
+outbreak_size_list <- lapply(prop, as.data.frame)
+for (i in seq_along(offspring_dists)) {
+  outbreak_size_list[[i]]$R <- get_parameters(offspring_dists[[i]])[["mean"]]
+  outbreak_size_list[[i]]$k <- get_parameters(offspring_dists[[i]])[["dispersion"]]
+  outbreak_size_list[[i]]$offspring_dist <- family(offspring_dists[[i]])
+  outbreak_size_list[[i]]$disease <- diseases[i]
+  outbreak_size_list[[i]]$statistic <- "size"
+}
+outbreak_size <- do.call(rbind, outbreak_size_list)
```

Comment:
Add more detail to the disease names to make it easier to interpret the plots
```suggestion
outbreak_size3 <- do.call(rbind, outbreak_size_list)

outbreak_size3 <- outbreak_size3 |>
  dplyr::mutate(
    disease = stringr::str_remove(disease, "\\.\\d+$"),
    interval = factor(interval, levels = unique(interval)),  # preserve order
    # Create a label combining R and k for each disease
    disease_label = paste0(disease, "\n(R=", R, ", k=", k, ")")
  )
```